snaploader/src/main/java/electrostatic4j/snaploader/platform/util/NativeVariant.java

```diff
@@ -250,6 +260,46 @@ public static boolean isAMD() {
         public static boolean isARM() {
             return OS_ARCH.getProperty().contains("arm") || OS_ARCH.getProperty().contains("aarch");
         }
+
+        /**
+         * Tests whether the named ISA extensions are all present.
+         *
+         * @param requiredNames the names of the extensions to test for
+         * @return {@code true} if all are present, otherwise {@code false}
+         */
+        public static boolean hasExtensions(String... requiredNames) {
+            // Obtain the list of CPU feature strings from OSHI:
+            SystemInfo si = new SystemInfo();
+            HardwareAbstractionLayer hal = si.getHardware();
+            CentralProcessor cpu = hal.getProcessor();
+            List<String> oshiList = cpu.getFeatureFlags();
+
+            Pattern pattern = Pattern.compile("[a-z][a-z0-9_]*");
+
+            // Convert the list to a collection of feature names:
+            Collection<String> presentFeatures = new TreeSet<>();
+            for (String oshiString : oshiList) {
+                String lcString = oshiString.toLowerCase(Locale.ROOT);
+                Matcher matcher = pattern.matcher(lcString);
+                while (matcher.find()) {
+                    String featureName = matcher.group();
+                    presentFeatures.add(featureName);
+                }
+            }
+
+            // Test for each required extension:
+            for (String extensionName : requiredNames) {
+                String lcName = extensionName.toLowerCase(Locale.ROOT);
+                String pfName = "pf_" + lcName + "_instructions_available";
```

Comment:
Since, we are going to continue with the pattern mathcing solution which deemed the best for now. I need to understand this part. If you can, kindly place a comment describing why appending a prefix and postfix here. Thanks.

Reply:
Done.